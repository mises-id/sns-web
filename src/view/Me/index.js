/*
 * @Author: lmk
 * @Date: 2021-07-08 15:08:05
 * @LastEditTime: 2021-07-15 16:42:50
 * @LastEditors: lmk
 * @Description: 
 */
import React from 'react';
import './index.scss'
import { useTranslation } from 'react-i18next';
import following from '@/images/following.png'
import followers from '@/images/followers.png'
import post from '@/images/post.png'
import Cell from '@/components/Cell';
const Myself = ({history})=>{
  const {t} = useTranslation()
  const list = [{
    label:t('following'),
    icon:following,
    url:'/follow',
    pageType:'following'
  },{
    label:t('followers'),
    icon:followers,
    url:'/follow',
    pageType:'followers'
  },{
    label:t('posts'),
    icon:post,
    url:''
  }]
  const userInfo = ()=>{
    history.push('/userInfo')
  }
  const cellClick = val=>{
    history.push({pathname:val.url,state:{pageType:val.pageType}})
  }
  return <div className="m-layout">
    <div className="m-padding-top15  m-bg-f8f8f8"></div>
    <div className="m-padding-lr15">
      <Cell iconSize={60}  label="Susan" onPress={userInfo}></Cell>
      {list.map((val,index)=>(<Cell showLine={false} icon={val.icon} iconSize={20} key={index}  label={val.label} onPress={()=>cellClick(val)}></Cell>))}
    </div>
  </div>
}
export default Myself