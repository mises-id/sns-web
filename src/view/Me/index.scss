.nickname{
  @include title(25)
}
.footer{
  padding: 0 40px;
  .za-button--md{
    height: 50px;
  }
}
.bg{
  display: block;
  width: 100%;
}
.m-tips{
  word-break: keep-all;
  word-wrap: break-word; 
  white-space: pre-wrap
}