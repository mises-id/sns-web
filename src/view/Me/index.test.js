import React from 'react';
import Me from './';
import {shallow } from 'enzyme';
describe('Me', () => {
  it('should render correctly', () => {
    const component = shallow(<Me />)
    expect(component).toMatchSnapshot();
  });
});