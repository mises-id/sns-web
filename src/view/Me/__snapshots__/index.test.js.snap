// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me should render correctly 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "20px",
        "textAlign": "center",
      }
    }
  >
    <ActivityIndicator
      loading={true}
      percent={20}
      prefixCls="za-activity-indicator"
      strokeWidth={5}
      type="spinner"
    />
  </div>
</div>
`;
